chore: add analytics documentation (#515)
front-commerce/developers.front-commerce.com: docs/advanced/analytics/getting-started.mdx

---
sidebar_position: 1
title: Getting Started
description:
  Get started by learning how to track analytics events across your entire
  application.
---

import AnalyticsFigure from "./assets/analytics.svg";
import Figure from "@site/src/components/Figure";
import ContactLink from "@site/src/components/ContactLink";
import { useState } from "react";

> _Since version 2.19_

<p>{frontMatter.description}</p>

> Front-Commerce uses [`getanalytics.io`](https://getanalytics.io/) under the

Reviewer: **issue:**  the project name is Analytics  and library is named `analytics`. I think mentioning the project web domain name instead  of its name could lead to confusion

**suggestion:** use "Analytics" or "`analytics`"
```suggestion
Front-Commerce uses [`analytics`](https://getanalytics.io/) under the
```

hood. If we represent how it works, it would look like this:

<Figure>
  <AnalyticsFigure

>     aria-label="Schema explaining the concepts behind analytics.js"

Reviewer: :eye: 
```suggestion
    aria-label="Schema explaining the concepts behind analytics"
```

    width="100%"
    className="text-black dark:text-white" // the lines use current color as to adabt to theme
  />
</Figure>

Across your React application, you can track events using functions such as
`trackEvent` or `trackPage`. Then, the event is dispatched to all the relevant
integrations registered in your application, learn more in the
[Tracking API](tracking-api) docs.

This means that once you have correctly configured events in your React

> Components, adding new tracking can be really straightforward.

Reviewer: **nitpick:** it may not be _that_ straightforward for everyone.

**suggestion:** list the benefit·s in a more objective way
```suggestion
Components, adding new tracking services is less risky: it has no impact on what is being tracked.
```

_I'm not super fan of the suggestion rewording. Feel free to edit :angel:_

## Configure analytics

First you will need to configure your analytics by updating the
[configuration file](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/src/config/analytics.sample.js):

> 

Reviewer: **hypothesis:**  readers might want to directly switch to the available integrations so they know how to setup the plugin that makes sense for their task at hand

**suggestion:**  add a callout redirecting to /docs/advanced/analytics/plugins#common-plugins
